### DISCUSSION TITLE
HTTPS Exception - MIRTH 4.4.1

### DISCUSSION THREAD

**Discussion by boukenka** [nextgenhealthcare/connect#5961] [Frequently Asked Questions]
Hello,

Sorry to be back. But this time I have an issue with HTTPS.
There is a download of information from an API REST. Below is the code.

```
  var url = new java.net.URL('https://.....');
  var conn = url.openConnection();
  conn.setConnectTimeout(40000);
  conn.setReadTimeout(60000);
  conn.setDoOutput(true);
  conn.setRequestMethod('GET');
  conn.setRequestProperty('Accept', 'application/json');
  conn.setRequestProperty('Content-Type', 'application/json');
  conn.setRequestProperty("Authorization", 'XXXX...');
```

I get the following exception.

_ERROR  (js-connector:138): Process Invoicing web service Processing: Error: JavaException: java.lang.IllegalAccessException: class org.mozilla.javascript.MemberBox cannot access class sun.net.www.http.PosterOutputStream (in module java.base) because module java.base does not export sun.net.www.http to unnamed module @1eba372c_

I have launched MIRTH with the following command.

_docker run -e VMOPTIONS=--add-exports=java.base/sun.net.www.protocol.https=ALL-UNNAMED,--add-exports=java.base/sun.net.www.protocol.http=ALL-UNNAMED -p 8445:8443 nextgenhealthcare/connect_

Am I doing something wrong ? I guess it is the case 😄

**Comment by jonbartels** [comment]
Using java.net.URL is not wrong but using Apache HTTP Commons will work more reliably and without messing with JVM options. See an example here: https://github.com/nextgenhealthcare/connect/discussions/5157

**Reply by boukenka** [reply]
Thanks for your help. But for now, we have a lot of code with java.net.URL. I'd rather find a solution before migrating all templates to Apache HTTP Commons.

**Reply by boukenka** [reply]
We plan to migrate soon from MIRTH 3.5.1 to MIRTH 4.4.1.

**Reply by jonbartels** [reply]
What Java version are you using?

I think adjusting JVM args is the right direction. I am not sure what args would be needed.

The issue seems easy to reproduce. I'll try it when I need a brain break today.

**Reply by boukenka** [reply]
I guess it is Java 17 for MIRTH 4.4.1

**Comment by ab-mg-23** [comment]
Try `add-opens` instead of `add-exports`. Similar issue reported against Eclipse BIRT. https://github.com/eclipse-birt/birt/issues/1409

**Reply by boukenka** [reply]
Thanks @ab-mg-23 for the tip. Thanks also to @jonbartels . I have started to modify the code of my various libraries based on your example.
